### PR TITLE
add Support for -c class /path/to/main.ump

### PR DESCRIPTION
**Pull request by Digized**
This pr allows users to pass in a path to the umple file with the `-c` option.

### CODE REVIEW

File: cruise.umple/src/Compiler.ump [hunk 1]
```diff
@@ -43,18 +43,19 @@ class CodeCompiler {
...
   private static boolean compileJava(UmpleElement aClass, UmpleModel model, String args) {
     String path="";
+    String base="";
     for (GenerateTarget gt : model.getGenerates()) {
       if (gt.getLanguage().equals("Java")) {
-        path = StringFormatter.addPathOrAbsolute( 
-            model.getUmpleFile().getPath() +"/"+ gt.getPath(), "") + 
-          aClass.getPackageName().replace(".", File.separator);
+        base = StringFormatter.addPathOrAbsolute( 
+            model.getUmpleFile().getPath() +File.separator+ gt.getPath(), "");
```
Comment (Digized) [inline cruise.umple/src/Compiler.ump:50]: not sure why this was being used with `/` before

**Comment by TimLethbridge** [comment]
@Digized  This does not seem to have solved the problem I thought you were trying to solve. Can you give me a concrete example of what it actually solves? From the code it looks as though it is solving command-line use on Windows. But it doesn't seem to actually allow you to use the -c option with more than a single class when the directory is not the current directory.

I built this locally. Then I added an umple file hw.ump within directory a/b  (the second hello world example from https://cruise.eecs.uottawa.ca/umple/HelloWorldExamples.html). Then I tried the following

```
java -jar ~/umple/dist/umple.jar -c Person Student Mentor /Users/tcl/tmp/xt/a/b/hw.ump
```

I still get the following errors:

```
hw.ump:34: error: cannot find symbol
    Mentor m = new Mentor("Nick The Mentor");
    ^
  symbol:   class Mentor
  location: class Person
hw.ump:34: error: cannot find symbol
    Mentor m = new Mentor("Nick The Mentor");
                   ^
  symbol:   class Mentor
  location: class Person
hw.ump:35: error: cannot find symbol
    Student s = new Student("Tom The Student");
    ^
  symbol:   class Student
  location: class Person
hw.ump:35: error: cannot find symbol
    Student s = new Student("Tom The Student");
                    ^
  symbol:   class Student
  location: class Person
4 errors

```

I also created a .ump file with a single class that has a main. It works fine on a Mac, but so did the old version.

Almost all users use the '-' option instead of a single class by the way.

**Comment by Digized** [comment]
when I run something like `java -jar umple.jar -c Person ./umpletest/hw.ump` without this code change I get 
```
Processing -> ./umpletest/hw.ump
hw.ump:34: error: cannot find symbol
    Mentor m = new Mentor("Nick The Mentor");
    ^
  symbol:   class Mentor
  location: class Person
hw.ump:34: error: cannot find symbol
    Mentor m = new Mentor("Nick The Mentor");
                   ^
  symbol:   class Mentor
  location: class Person
hw.ump:35: error: cannot find symbol
    Student s = new Student("Tom The Student");
    ^
  symbol:   class Student
  location: class Person
hw.ump:35: error: cannot find symbol
    Student s = new Student("Tom The Student");
                    ^
  symbol:   class Student
  location: class Person
4 errors
Processed ./umpletest/hw.ump.
```

with the code change, I get a correct result. 

also I just tested out `java -jar umple.jar -c - ./umpletest/hw.ump` without the code change as well and it produces the same 4 errors.

with the code change I get the correct result. 

vscode has a path to the current file. when a user wants to compile  a project I ask them what class they want to compile, I dont expect them to add multiple files.

**Comment by TimLethbridge** [comment]
@Digized. I will merge it. My mistake. The general protocol has been to create branches in Umple itself rather than do PRs from a fork. It is OK that you did, but I used the wrong command so ended up building the default system rather than the actual branch.

**Comment by Digized** [comment]
Yea sorry about that, since I am not a member of the umple organization, I can't actually create branches on `umple/umple`

**Pull request merged**